GTBitsOfGood/fair-districts-ga
# Campaign sequence

## Campaign creation sequence

Issue Number(s): #26 

Automatically matches volunteers and newspapers:

- Matching guidelines
  - Newspapers in the targeted counties are sorted from highest rating to lowest rating (desc)
  - Volunteers are sorted primarily by county (if their associated county field is within the targeted counties) and then by number of existing assignments (asc) - volunteers with the least assignments will be more fit to take on new assignments (?)
- In the case that the number of volunteers exceeds the number of newspapers in the targeted counties:
  - The volunteers in the targeted counties are assigned first
  - The number of assignments made is equal to the number of newspapers
  - If it is decided that a user wants to assign multiple people to a newspaper, this **is** possible since there are more volunteers than newspapers
  - A "+" button to add assignment fields was added for this purpose
- For the case that the number of newspapers exceeds the number of volunteers in the targeted counties:
  - The number of newspapers assigned are restricted to the number of volunteers
  - The user can then select which newspapers to assign the volunteers since there are not enough volunteers to assign to newspapers

The UI also includes an "X" button to remove assignments as needed.

@hdemusg - this PR also introduces initial work for the individual campaign pages.

### Critical Changes

- Schema was updated. Ensure `npx prisma migrate dev` is run upon pulling these changes.

### How to Test

1. Create various newspapers and volunteers and note the counties they are assigned to (for testing you may want to create a different number of newspapers and volunteers, perhaps try newspapers > volunteers and then volunteers < newspapers)
2. Walk through the campaign sequence (both counties focus and legislator focus) and ensure that nothing breaks

### Next Steps
1. Volunteer page should show assigned newspapers
2. Individual campaign detail pages must show assignments
